src/main.rs

```diff
@@ -730,19 +754,26 @@ fn retrieve_and_validate_single_entry(
     record_db: &mut RecordDatabase,
     citation_key: RecordId,
     client: &HttpClient,
+    format: OutputFormat,
     ignore_null: bool,
 ) -> Result<Option<(Entry<RawRecordData>, RemoteId)>, error::Error> {
     match get_record_row(record_db, citation_key, client)? {
-        RecordRowResponse::Exists(record, row) => {
-            let Record {
-                key,
-                data,
-                canonical,
-            } = record;
-            let entry = validate_bibtex_entry(key, data, &row).map(|entry| (entry, canonical));
-            row.commit()?;
-            Ok(entry)
-        }
+        RecordRowResponse::Exists(record, row) => match format {
+            OutputFormat::Bibtex => {
+                let Record {
+                    key,
+                    data,
+                    canonical,
+                } = record;
+                let entry = validate_bibtex_entry(key, data, &row).map(|entry| (entry, canonical));
+                row.commit()?;
+                Ok(entry)
+            }
+            OutputFormat::Null => {
+                row.commit()?;
+                Ok(None)
+            }
+        },
```

Comment:
I don't really think this optimization is worthwhile, it increases the complexity of this implementation for a rather uncommon function call.

Reply:
What would you suggest we do? `validate_bibtex_entry` just handles the key validation, so I think it makes sense to skip it altogether.

Reply:
Oh I see, of course when the `Entry` is generated, the error is already cause.